### PR TITLE
Truncate topic description

### PR DESCRIPTION
**Pull request by christianmemije**
### Summary

Truncate topic description using the text-truncator component. Replaces the shave-text component.

localhost_8000_learn_

…

### Reviewer guidance

View a topic with a very long description.

----

### Contributor Checklist

- [x] Contributor has fully tested the PR manually
- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [x] If there are any front-end changes, before/after screenshots are included
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

### CODE REVIEW

**Comment by indirectlylit** [comment]
Currently this is driven by a max char limit. I wonder if it might be preferable to drive it based on a max height, as in the [shaved text component](https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/learn/assets/src/views/content-card/shaved-text.vue)? 

I can see pros and cons, but using max height (maybe 2 lines?) seems to be a little more responsive.

**Comment by indirectlylit** [comment]
(Note that you could give the `shaved-text` component `hasTooltip` and `hasShowMoreLink` props)

File: kolibri/plugins/coach/assets/src/views/lessons/LessonContentPreviewPage/MetadataArea.vue [hunk 1]
```diff
@@ -108,7 +109,10 @@
         return false;
       },
       description() {
-        return this.content.description;
+        if (this.content) {
```
Comment (christianmemije) [inline kolibri/plugins/coach/assets/src/views/lessons/LessonContentPreviewPage/MetadataArea.vue:112]: Was missing this.

File: kolibri/core/assets/src/views/text-truncator.vue [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      :title="tooltipText"
+    >
+      {{ text }}
+    </div>
+    <div class="ar">
+      <k-button
+        v-if="showViewMore && (textIsTruncated || viewAllText)"
+        appearance="basic-link"
+        :text="viewAllText ? $tr('viewLessButtonPrompt') : $tr('viewMoreButtonPrompt')"
+        @click.stop.prevent="viewAllText =! viewAllText"
```
Comment (jonboiser) [inline kolibri/core/assets/src/views/text-truncator.vue]: `        @click.stop.prevent="viewAllText = !viewAllText"` here

Comment (christianmemije) [reply kolibri/core/assets/src/views/text-truncator.vue]: Updated @jonboiser

**Pull request merged**